itest/assets_test.go

```diff
@@ -473,3 +473,31 @@ func testMintAssetsWithTapscriptSibling(t *harnessTest) {
 	t.lndHarness.MineBlocksAndAssertNumTxes(1, 1)
 	t.lndHarness.AssertNumUTXOsWithConf(t.lndHarness.Bob, 1, 1, 1)
 }
+
+// testMintAssetWithPrecision tests that we're able to mint an asset with a
+// specific precision.
+func testMintAssetWithPrecision(t *harnessTest) {
+	mintName := "test-asset-precision"
+	jsonData := []byte(`{"field1": "value1", "field2": "value2"}`)
+
+	assetMint := mintrpc.MintAssetRequest{
+		Asset: &mintrpc.MintAsset{
+			AssetType: taprpc.AssetType_NORMAL,
+			Name:      mintName,
+			AssetMeta: &taprpc.AssetMeta{
+				Data: jsonData,
+				Type: taprpc.AssetMetaType_META_TYPE_JSON,
+			},
+			Amount: 500,
+			// Precision: 2,
```

Comment:
this is temp as currently the itest seems to be reporting mismatching metahashes between the generated metadata hash and what is present in the genesis info

Reply:
there was a hard-coded `TypeOpaque` metadata type in an itest helper method, this is fixed now